topics-project/__tests__/signin-form.test.js

```diff
@@ -0,0 +1,93 @@
+/** @jest-environment jsdom */
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import LoginPage from "../app/login/page";
+import { createClient } from "../src/utils/supabase/client";
+import { login } from "../app/login/actions";
+
+jest.mock("../src/utils/supabase/client", () => ({
+  createClient: jest.fn().mockReturnValue({
+    auth: {
+      signInWithPassword: jest.fn().mockResolvedValue({ error: null }), // Simulate success
+    },
+  }),
+}));
+
+jest.mock("../app/login/actions", () => ({
+  login: jest.fn().mockResolvedValue({ message: "success", error: null }), // Mock login function
+}));
+
+describe("Sign-in form", () => {
+  const user = userEvent.setup();
+  it("Form renders properly", () => {
+    render(<LoginPage />);
+
+    const emailField = screen.getByPlaceholderText("Enter email");
+    const passwordField = screen.getByPlaceholderText("Enter password");
+
+    expect(emailField).toBeInTheDocument();
+    expect(passwordField).toBeInTheDocument();
+  });
+
+  it("Form email field requires email formatting", async () => {
+    render(<LoginPage />);
+
+    const emailField = screen.getByPlaceholderText("Enter email");
+
+    await user.type(emailField, "johan");
+    expect(emailField).toBeInvalid();
+
+    await user.type(emailField, "test@gmail.com");
+    expect(emailField).toBeValid();
+  });
+  // Integration Tests are the two below
```

Comment:
Different types of tests need to be separated into different files. Additionally, tests of different components also need to be separated into different files. This makes it simpler to diagnose test failures.